core/src/main/java/org/elasticsearch/rest/action/cat/RestThreadPoolAction.java

```diff
@@ -149,96 +96,39 @@ public RestResponse buildResponse(NodesStatsResponse nodesStatsResponse) throws
-            );
-        }
-
+        table.addCell("name", "default:true;alias:n;desc:thread pool name");
+        table.addCell("type", "alias:t;default:false;desc:thread pool type");
+        table.addCell("active", "alias:a;default:true;text-align:right;desc:number of active threads");
+        table.addCell("size", "alias:s;default:false;text-align:right;desc:number of threads");
+        table.addCell("queue", "alias:q;default:true;text-align:right;desc:number of tasks currently in queue");
+        table.addCell("queue_size", "alias:qs;default:false;text-align:right;desc:maximum number of tasks permitted in queue");
+        table.addCell("rejected", "alias:r;default:true;text-align:right;desc:number of rejected tasks");
+        table.addCell("largest", "alias:l;default:false;text-align:right;desc:highest number of seen active threads");
+        table.addCell("completed", "alias:c;default:false;text-align:right;desc:number of completed tasks");
+        table.addCell("min", "alias:mi;default:false;text-align:right;desc:minimum number of threads");
+        table.addCell("max", "alias:ma;default:false;text-align:right;desc:maximum number of threads");
+        table.addCell("keep_alive", "alias:ka;default:false;text-align:right;desc:thread keep alive time");
         table.endHeaders();
         return table;
     }
 
-
     private Table buildTable(RestRequest req, ClusterStateResponse state, NodesInfoResponse nodesInfo, NodesStatsResponse nodesStats) {
-        boolean fullId = req.paramAsBoolean("full_id", false);
-        DiscoveryNodes nodes = state.getState().nodes();
-        Table table = getTableWithHeader(req);
+        final String[] threadPools = req.paramAsStringArray("thread_pools", new String[] { ".*" });
+        final DiscoveryNodes nodes = state.getState().nodes();
+        final Table table = getTableWithHeader(req);
 
-        for (DiscoveryNode node : nodes) {
-            NodeInfo info = nodesInfo.getNodesMap().get(node.getId());
-            NodeStats stats = nodesStats.getNodesMap().get(node.getId());
-            table.startRow();
-
-            table.addCell(fullId ? node.getId() : Strings.substring(node.getId(), 0, 4));
-            table.addCell(info == null ? null : info.getProcess().getId());
-            table.addCell(node.getHostName());
-            table.addCell(node.getHostAddress());
-            if (node.getAddress() instanceof InetSocketTransportAddress) {
-                table.addCell(((InetSocketTransportAddress) node.getAddress()).address().getPort());
-            } else {
-                table.addCell("-");
-            }
+        for (final DiscoveryNode node : nodes) {
+            final NodeInfo info = nodesInfo.getNodesMap().get(node.getId());
+            final NodeStats stats = nodesStats.getNodesMap().get(node.getId());
 
             final Map<String, ThreadPoolStats.Stats> poolThreadStats;
```

Comment:
+1, I will add such a comment. Also, one of the REST tests will catch a regression here since I test multiple unordered thread pool names and ensure that the output is ordered by name.